test/CMakeLists.txt

```diff
@@ -2,12 +2,14 @@ include(FetchContent)
 
 set(DOCTEST_NO_INSTALL ON)
 FetchContent_Declare(doctest
-    GIT_REPOSITORY "https://github.com/doctest/doctest.git"
+    GIT_REPOSITORY https://github.com/doctest/doctest.git
     GIT_TAG v2.4.9
 )
 FetchContent_MakeAvailable(doctest)
-list(APPEND CMAKE_MODULE_PATH ${doctest_SOURCE_DIR}/scripts/cmake)
-include(doctest)
+set_target_properties(doctest_with_main PROPERTIES COMPILE_OPTIONS "" EXPORT_COMPILE_COMMANDS OFF)
+get_target_property(DOCTEST_INCLUDE_DIRS doctest INTERFACE_INCLUDE_DIRECTORIES)
+target_include_directories(doctest SYSTEM INTERFACE ${DOCTEST_INCLUDE_DIRS})
```

Comment:
A small comment explaining why those extra step are needed would be great.

Reply:
I added a few comments for this since it is rather unintuitive.